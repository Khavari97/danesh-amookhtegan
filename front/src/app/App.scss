@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


