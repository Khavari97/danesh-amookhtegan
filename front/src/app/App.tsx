import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
