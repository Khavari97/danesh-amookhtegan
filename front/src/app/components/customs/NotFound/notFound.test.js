import React from "react";
import Enzyme, { shallow, configure, mount } from "enzyme";
import renderer from "react-test-renderer";
import App from './NotFound';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Typography, Grid } from '@material-ui/core';

configure({ adapter: new Adapter() })


describe('notfound component', () => {
  let wrapper = mount(<App />)
  it('render component', () => {
    expect(wrapper.find(Typography).exists()).toBe(true);
    expect(wrapper.find(Typography).length).toEqual(1);
    expect(wrapper.find(Grid).exists()).toBe(true);
    expect(wrapper.find(Grid).length).toEqual(1);
  })
  it('check default props', () => {
    expect(wrapper.find(Typography).html()).toBe("<p class=\"MuiTypography-root MuiTypography-body1\" style=\"margin-right: 10px;\"></p>");
    expect(wrapper.find(Grid).props().alignItems).toEqual("center");
    expect(wrapper.find(Grid).props().alignItems).toEqual("center");
  })

  it('check set props', () => {
    wrapper.setProps({ label: 'test' })
    expect(wrapper.props().label).toBe('test')
    expect(wrapper.find(Typography).html()).toBe("<p class=\"MuiTypography-root MuiTypography-body1\" style=\"margin-right: 10px;\">test</p>");
  })
})

describe('render NotFound component without error', () => {
  let wrapper = shallow(<App />)

  it('it should have value', () => {
    wrapper.setProps({
      label: 'string'
    })
    expect(wrapper.find(Typography).text()).toBe('string');
  });
      it('it should have value', () => {
          wrapper.setProps({
              label: 'string'
          })
         expect(wrapper.find(Typography).text()).toBe('string');
      });

    it('snapshot',()=>{
        const shot = <App />
        const snapshot = renderer.create(shot).toJSON();
        expect(snapshot).toMatchSnapshot();
    })

})