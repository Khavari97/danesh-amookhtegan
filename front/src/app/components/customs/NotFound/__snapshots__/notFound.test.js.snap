// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render NotFound component without error snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body1"
    style={
      Object {
        "marginRight": "10px",
      }
    }
  />
</div>
`;
